hermes-management/src/main/java/pl/allegro/tech/hermes/management/infrastructure/kafka/service/KafkaBrokerTopicManagement.java

```diff
@@ -1,95 +1,126 @@
-                )
+                kafkaAdminClient.createTopics(Collections.singletonList(
+                        new NewTopic(
+                                k.name().asString(),
+                                topicProperties.getPartitions(),
+                                (short) topicProperties.getReplicationFactor()
+                        ).configs(config)
+                ))
         );
     }
 
     @Override
     public void removeTopic(Topic topic) {
-        kafkaNamesMapper.toKafkaTopics(topic).forEach(k -> adminZkClient.deleteTopic(k.name().asString()));
+        kafkaNamesMapper.toKafkaTopics(topic).forEach(k -> kafkaAdminClient.deleteTopics(Collections.singletonList(k.name().asString())));
     }
 
     @Override
     public void updateTopic(Topic topic) {
-        Properties config = createTopicConfig(topic.getRetentionTime().getDuration(), topicProperties);
+        Map<String, String> config = createTopicConfig(topic.getRetentionTime().getDuration(), topicProperties);
         KafkaTopics kafkaTopics = kafkaNamesMapper.toKafkaTopics(topic);
 
         if (isMigrationToNewKafkaTopic(kafkaTopics)) {
-            adminZkClient.createTopic(
-                    kafkaTopics.getPrimary().name().asString(),
-                    topicProperties.getPartitions(),
-                    topicProperties.getReplicationFactor(),
-                    config,
-                    kafka.admin.RackAwareMode.Enforced$.MODULE$
-            );
+            kafkaAdminClient.createTopics(Collections.singletonList(
+                    new NewTopic(
+                            kafkaTopics.getPrimary().name().asString(),
+                            topicProperties.getPartitions(),
+                            (short) topicProperties.getReplicationFactor()
+                    ).configs(config)
+            ));
         } else {
-            adminZkClient.changeTopicConfig(kafkaTopics.getPrimary().name().asString(), config);
+            doUpdateTopic(kafkaTopics.getPrimary(), config);
         }
 
         kafkaTopics.getSecondary().ifPresent(secondary ->
-                adminZkClient.changeTopicConfig(secondary.name().asString(), config)
+                doUpdateTopic(secondary, config)
         );
     }
 
     @Override
     public boolean topicExists(Topic topic) {
         return kafkaNamesMapper.toKafkaTopics(topic)
-                .allMatch(kafkaTopic -> kafkaZkClient.topicExists(kafkaTopic.name().asString()));
+                .allMatch(this::doesTopicExist);
     }
 
     private boolean isMigrationToNewKafkaTopic(KafkaTopics kafkaTopics) {
         return kafkaTopics.getSecondary().isPresent() &&
-                !kafkaZkClient.topicExists(kafkaTopics.getPrimary().name().asString());
+                !doesTopicExist(kafkaTopics.getPrimary());
+    }
+
+    private boolean doesTopicExist(KafkaTopic topic) {
+        try {
+            return kafkaAdminClient.listTopics().names()
```

Comment:
I have mixed feelings about this one. An alternative to fetching all topics is to try to 'describe' topic and then catch an exception when it does not exist - I've decided to go with listing all topics, but I'm open to debate on this.

Reply:
Listing over Kafka cluster with more than 1k topics is faster than describing a topic. Therefore I would stay with your current implementation 

```
time ./kafka_2.12-2.6.0/bin/kafka-topics.sh --list --bootstrap-server nodeAddressWithOver1000Topics 
...
________________________________________________________
Executed in    2,64 secs   fish           external
   usr time    2,19 secs  147,00 micros    2,19 secs
   sys time    0,36 secs  722,00 micros    0,36 secs
```

vs.

```
time ./kafka_2.12-2.6.0/bin/kafka-topics.sh --describe --topic someTopicName --bootstrap-server nodeAddressWithOver1000Topics 
...
________________________________________________________
Executed in    2,97 secs   fish           external
   usr time    2,48 secs  153,00 micros    2,48 secs
   sys time    0,38 secs  757,00 micros    0,38 secs
```